br/pkg/checkpoint/checkpoint.go

```diff
@@ -240,7 +240,7 @@ func StartCheckpointRunnerForTest(ctx context.Context, storage storage.ExternalS
 
 		appendCh: make(chan *CheckpointMessage),
 		metaCh:   make(chan map[string]*RangeGroups),
-		errCh:    make(chan error),
```

Comment:
is the count 1 of buffer length enough to deal this issue, if the multiple error is sent to errCh ?
Can we close the errCh before receiver is closed ?

Reply:
The error incurs only from the flush stage, which is wrapped in a goroutine. Once an error incurs, this goroutine send back the error and exit, so no error will occur again.